// riot-valorant-api/dist/api/riotValorantApi.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RiotValorantApi = void 0;
const axios_1 = __importDefault(require("axios"));
const axiosInterceptor_1 = require("../utils/axiosInterceptor");
const rankedV1_1 = require("./valorant/ranked/rankedV1");
const contentV1_1 = require("./valorant/content/contentV1");
const matchV1_1 = require("./valorant/match/matchV1");
const statusV1_1 = require("./valorant/status/statusV1");
class RiotValorantApi {
    constructor(apiToken, region) {
        if (!apiToken) {
            throw new Error("Empty api token");
        }
        if (!region) {
            throw new Error("Empty region");
        }
        this.region = region;
        const axiosInstance = axios_1.default.create({
            baseURL: `https://${this.region}.api.riotgames.com/`,
            headers: {
                "X-Riot-Token": apiToken,
            },
        });
        axiosInstance.interceptors.response.use(axiosInterceptor_1.responseInterceptor, axiosInterceptor_1.errorInterceptor);
        this.RankedV1 = new rankedV1_1.RankedV1(axiosInstance);
        this.StatusV1 = new statusV1_1.StatusV1(axiosInstance);
        this.MatchV1 = new matchV1_1.MatchV1(axiosInstance);
        this.ContentV1 = new contentV1_1.ContentV1(axiosInstance);
    }
}
exports.RiotValorantApi = RiotValorantApi;
